framework/src/modules/interoperability/base_cross_chain_update_command.ts

```diff
@@ -603,4 +595,33 @@ export abstract class BaseCrossChainUpdateCommand<
 			return false;
 		}
 	}
+
+	protected async verifyCertificateSignatureAndPartnerChainOutboxRoot(
+		context: CommandExecuteContext<CrossChainUpdateTransactionParams>,
+	) {
+		const { params, transaction } = context;
+		const { inboxUpdate } = params;
+		// Verify certificate signature. We do it here because if it fails, the transaction fails rather than being invalid.
+		await this.internalMethod.verifyCertificateSignature(context, params);
+
+		if (!isInboxUpdateEmpty(inboxUpdate)) {
+			// This check is expensive. Therefore, it is done in the execute step instead of the verify
```

Comment:
maybe this comment could be moved to the top of the function to refer to both checks, as the rationale is similar